created SFTPBatchOperator which add batch function (#19852)
mistercrunch/Airflow: docs/apache-airflow-providers-sftp/sftp.rst

Use ssh_conn_id to specify the name of the connection.

You can use the operator for the following tasks:

1. Send one file to the server with the full path

.. code-block:: python

    put_file = SFTPOperator(
        task_id="put_file",
        ssh_conn_id="ssh_default",
        local_filepath="/tmp/transfer_file/put_file_file1.txt",
        remote_filepath="/tmp/transfer_file/remote/put_file_file1.txt",
        operation=SFTPOperation.PUT,
        create_intermediate_dirs=True,
    )

2. Send all files from local directory to remote server

.. code-block:: python

    put_dir_files = SFTPBatchOperator(
        task_id="put_dir_files",
        ssh_conn_id="ssh_default",
        local_path="/tmp/dir_for_remote_transfer/",
        remote_path="/tmp/dir_for_remote_transfer/remote/",
        operation=SFTPOperation.PUT,
        create_intermediate_dirs=True,
    )

3. Send all files from local directory to remote server

.. code-block:: python

    put_dir_files = SFTPBatchOperator(
        task_id="put_dir_files",
        ssh_conn_id="ssh_default",
        local_path="/tmp/dir_for_remote_transfer/",
        remote_path=[

>             "/tmp/dir_for_remote_transfer/remote/txt/file1.txt",

Reviewer: How would that work? I think this example makes no sense.

Author: Yes, i makes some mistake with this example, I fix that

Author: done

            "/tmp/dir_for_remote_transfer/remote/txt/file2.txt",
        ],
        operation=SFTPOperation.PUT,
        create_intermediate_dirs=True,
    )

4. Send all files from the local directory that match the specified pattern to the remote server

.. code-block:: python

    put_dir_txt_files = SFTPBatchOperator(
        task_id="put_dir_txt_files",
        ssh_conn_id="ssh_default",
        local_path="/tmp/dir_for_remote_transfer/",
        remote_path="/tmp/dir_for_remote_transfer/remote/txt/",

>         regexp_mask=".*[.]txt",

Reviewer: ```suggestion
        regexp_mask=r".*\.txt",
```

        operation=SFTPOperation.PUT,
        create_intermediate_dirs=True,
    )

5. Send list of files from the local directory that match the specified pattern to the remote server

.. code-block:: python

    put_dir_txt_files = SFTPBatchOperator(
        task_id="put_dir_txt_files",
        ssh_conn_id="ssh_default",
        local_path=[
            "/tmp/dir_for_remote_transfer/file1.txt",
            "/tmp/dir_for_remote_transfer/file2.txt",
        ],
        remote_path="/tmp/dir_for_remote_transfer/remote/txt/",

>         regexp_mask=".*[.]txt",

Reviewer: ```suggestion
        regexp_mask=r".*\.txt",
```

        operation=SFTPOperation.PUT,
        create_intermediate_dirs=True,
    )

The operator also supports transfer files from a remote server to a local,
for this you need to change the parameter ``operation`` from ``SFTPOperation.PUT`` to ``SFTPOperation.GET``.

> Parameter ``create_intermediate_dirs`` needed for create missing intermediate directories when

Reviewer: ```suggestion
Parameter ``create_intermediate_dirs`` is needed to create missing intermediate directories when
```

 .. Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

 ..   http://www.apache.org/licenses/LICENSE-2.0

 .. Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.

SFTPOperator
==========================
Use the :class:`~airflow.providers.sftp.operators.sftp.py` to
transfer data between servers under sftp.

Using the Operator
------------------
To start working with an operator, you need to register an SFTP \ SSH connection in Airflow Connections.

> Use ssh_conn_id to specify the name of the connection.

Reviewer 2: Is this doc meant to describe both `SFTPOperator` and `SFTPBatchOperator` or just one of the operators? This sections reads like the doc is about `SFTPOperator` only but there are code snippets for `SFTPBatchOperator` as well further on.

Author: this doc for a both operators

Author: done, i separated docs

Reviewer 2: I don't necessarily think you need to split the documentation especially since they are quite similar. This comment was really about making sure the documentation stated that is was about both operators rather than just `SFTPOperator`. It's very common to have one document for a grouping of operators and IMO this is a great candidate.
